### PR TITLE
Get assets from Core

### PR DESCRIPTION
**Pull request by AntonNeld**
Components now accept 'file' messages, containing assets that shall be written do the 'resources' directory.

Files are base64-encoded and sent in a single websocket message. This currently does not handle long videos well, which will be fixed later.

### CODE REVIEW

File: screen/model/fileHandler.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+    constructor(resourcesPath) {
+        this.resourcesPath = resourcesPath;
+    }
+
+    async writeFile(message) {
+        const { path: filePath, data } = message;
+        const fullPath = path.join(this.resourcesPath, filePath);
+        const dir = path.dirname(fullPath);
+        mkdir(dir, { recursive: true });
```
Comment (mzlarsson) [inline screen/model/fileHandler.js]: Do we need to await this?

**Review by mzlarsson — CHANGES_REQUESTED** [review]
Btw, could we set the resource path to be shared amongst components? E.g. if we have a video with sound both screen and audio needs to have access to it. Theoretically they could be on different devices but if they are on the same it would be a waste to keep two copies.

File: screen/model/fileHandler.js (hunk 1 above)
Comment (AntonNeld) [reply screen/model/fileHandler.js]: Yes, good catch

**Comment by AntonNeld** [comment]
Sharing the resource path could be tricky if both components are trying to write the files at the same time. It's worth looking into, but maybe as a later step?

**Review by mzlarsson — APPROVED** [review]

**Pull request merged**